Upload dockerd socket mount detection rule and test file

r0075h3ll opened the pull request:
This PR adds a new rule, along with a test file, for dockerd unix socket file i.e. `docker.sock` mount detection. 

#### Description

Docker works in client-server architecture. In a normal setup, `docker` command issued in the CLI interacts with the docker server(`dockerd`) using a Remote API or Unix Socket File(`docker.sock`) - the latter is often encountered. However, in a scenario where an attacker breaks into the container that has `/var/run/docker.sock` mounted, it can be leveraged to execute arbitrary commands on the host machine i.e. to talk back to the host machine thereby providing a way for escaping the container.

#### Example

Here's a dummy Dockerfile that mounts docker.sock file:
```Dockerfile
FROM docker:latest

WORKDIR /app

VOLUME /var/run/docker.sock:/var/run/docker.sock

CMD ["docker", "images"]
```

Upon running Semgrep with the provided rule, this is how the output looks like: 
<img width="881" alt="image" src="https://github.com/semgrep/semgrep-rules/assets/38760238/b90ffa01-a42d-4103-a39c-feac59a75882">

Thanks.

r0075h3ll committed:
Upload dockerd socket mount detection rule and test file

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3360) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3360) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/semgrep/semgrep-rules?pullRequest=3360) it.</sub>

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3360) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/semgrep/semgrep-rules?pullRequest=3360) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/semgrep/semgrep-rules?pullRequest=3360) it.</sub>

ievans commented:
thanks for the contribution! can you add a positive and negative test case annotation to the code? cf https://semgrep.dev/docs/writing-rules/testing-rules/

r0075h3ll committed:
Update dockerd-socket-mount.dockerfile

r0075h3ll commented:
Done :)

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

0xDC0DE committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll commented:
Hey @0xDC0DE, can we merge?

r0075h3ll committed:
Merge branch 'semgrep:develop' into develop

r0075h3ll committed:
Update documentbuilderfactory-disallow-doctype-decl-missing.yaml

Update the rule for checking if FEATURE_SECURE_PROCESSING is set to TRUE for DocumentBuilderFactory object.

r0075h3ll committed:
Merge branch 'semgrep:develop' into develop

p4p3r commented on java/lang/security/audit/xxe/documentbuilderfactory-disallow-doctype-decl-missing.yaml:
```diff
@@ -94,62 +97,64 @@ rules:
       - by-side-effect: true
         pattern-either:
           - patterns:
-            - pattern-either:
```
I think you inadvertently pushed changes to this rule which is unrelated to your PR. Could you please revert?

r0075h3ll replied:
My bad. I reverted the changes.

r0075h3ll committed:
Revert "Update documentbuilderfactory-disallow-doctype-decl-missing.yaml"

This reverts commit c1e22817279cfe141717e80458f48f7eff8ca71a.

r0075h3ll committed:
Merge branch 'develop' into develop

p4p3r reviewed (approved):
lgtm

r0075h3ll committed:
Merge branch 'develop' into develop

r0075h3ll committed:
Merge branch 'develop' into develop

0xDC0DE committed:
Merge branch 'develop' into develop

aryx commented on dockerfile/security/dockerd-socket-mount.yaml:
```diff
@@ -0,0 +1,36 @@
+        to escape container and execute arbitrary commands on the host machine.
+    languages:
+      - dockerfile
+      - yaml
```
@0xDC0DE why we have yaml listed as a language here?
